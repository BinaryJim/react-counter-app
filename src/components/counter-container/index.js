export { CounterContainer } from "./counter-container"