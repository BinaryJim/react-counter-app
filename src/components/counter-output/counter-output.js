import React from "react";

export const CounterOutput = () => {
  return (
    <section className="counter-output">
      <span className="counter-output__count">0</span>
    </section>
  );
};
