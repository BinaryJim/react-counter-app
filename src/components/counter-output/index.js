export { CounterOutput } from "./counter-output";
