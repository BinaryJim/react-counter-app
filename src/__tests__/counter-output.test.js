import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { CounterOutput } from "../components/counter-output";

jest.mock("../store", () => ({
  useCounterStateContext: jest.fn().mockReturnValue(0)
}))

// Test render
describe("CounterOutput - test render", () => {
  it("should render without crashing", () => {
    shallow(<CounterOutput />);
  });
});

// Testing output of the render
describe("CounterOutput - render", () => {
  let component;

  beforeAll(() => {
    component = shallow(<CounterOutput />)
  })

  it("should match snapshot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  
  it('should output the correct count value passed in by context', () => {
    expect(component.find('.counter-output__count').text()).toEqual("0");
  })
})