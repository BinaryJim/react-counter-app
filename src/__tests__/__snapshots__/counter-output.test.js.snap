// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterOutput - render should match snapshot 1`] = `
<section
  className="counter-output"
>
  <span
    className="counter-output__count"
  >
    0
  </span>
</section>
`;
